### ISSUE TITLE
[SessionManager Logs] Generated bucket policy of central logs bucket doesn't allow storage 

### ISSUE THREAD

**Issue by richardkeit** [awslabs/landing-zone-accelerator-on-aws#340]
**Describe the bug**
Launching an instance configured with generated EC2 Instance profile cannot store logs in central logs bucket

**To Reproduce**
Launch solution with below configuration:
_iam-config.yaml_:
~~~yaml
roleSets:
  - deploymentTargets:
      organizationalUnits:
        - Root
    roles:
      - name: EC2-Default-SSM-AD-Role
        instanceProfile: true
        assumedBy:
          - type: service
            principal: ec2.amazonaws.com
        policies:
          awsManaged:
            - AmazonSSMManagedInstanceCore
            - AmazonSSMDirectoryServiceAccess
            - CloudWatchAgentServerPolicy
~~~

_global-config.yaml_:
~~~yaml
logging:
  account: LogArchive
  sessionManager:
    sendToCloudWatchLogs: false
    sendToS3: true
    excludeRegions: []
    excludeAccounts: []
    lifecycleRules: []
    attachPolicyToIamRoles:
      - EC2-Default-SSM-AD-Role
~~~

- Launch EC2 instance
- Connect via SSM system manager
- Complete any actions on the EC2 instance
- Terminate session
- Session logs not found in central logs bucket

**Expected behavior**
Session logs found in `s3://aws-accelerator-central-logs-<SOLUTION_LOGGING_ACCOUNT>-<SOLUTION_HOME_REGION>/session/<ACCOUNT_ID>/<REGION>/<log>`

**Please complete the following information about the solution:**

- [x] Version: v1.5.2
- [x] Region: ap-southeast-2
- [ ] Was the solution modified from the version published on this repository? 
    - No
- [ ] Have you checked your [service quotas](https://docs.aws.amazon.com/general/latest/gr/aws_service_limits.html) for the services this solution uses? 
    - No
- [ ] Were there any errors in the CloudWatch Logs?
    - No
**Screenshots**
If applicable, add screenshots to help explain your problem (please **DO NOT include sensitive information**).

**Additional context**
As the generated bucket policy has the below condition, roles not matching the Accelerator prefix cannot place logs in the bucket:
~~~json
            "Condition": {
                "StringEquals": {
                    "aws:PrincipalOrgID": "<org_id>"
                },
                "ArnLike": {
                    "aws:PrincipalARN": [
                        "arn:aws:iam::*:role/AWSAccelerator-*",
                        "arn:aws:iam::*:role/cdk-accel-*",
                        "arn:aws:iam::*:role/AWSControlTowerExecution-*"
                    ]
                }
            }
~~~

The `logging.sessionManager.attachPolicyToIamRoles` therefore is misleading as it's implied that the session logs "will just magically get there".

**Comment by bhkhatri221** [comment]
Hi @richardkeit, thank you for filing an issue with the Landing Zone Accelerator team! 

There cloud be multiple reasons that might cause the session Manager logging to Amazon S3 as mentioned in the [How do I troubleshoot Session Manager logging to Amazon S3 or CloudWatch?](https://repost.aws/knowledge-center/ssm-troubleshoot-session-manager-logging). Can you kindly verify either internet gateway or gateway endpoint is present in the VPC where instance are hosted ? Having one of those configured in the VPC should enable the necessary connectivity for session manager logging to work properly.

Let me know the results once you have been able to configured the above requirements. Thank you for supporting Landing Zone Accelerator!

**Comment by richardkeit** [comment]
Hi @bhkhatri221 ,

It is 100% the bucket policy blocked access as changing (from my above example) the role from `EC2-Default-SSM-AD-Role` to `AWSAccelerator-EC2-Default-SSM-AD-Role` (where `AWSAccelerator-` is the `AcceleratorPrefix` from the InstallerStack).

Please read the `Additional context` header, as the root cause is explained there.
Let me know if any further explanation is required

**Comment by bhkhatri221** [comment]
Hi @richardkeit , Thank you for highlighting the additional context section. I understand the the root cause now. With the behavior you outlined, I've gone ahead and filed a bug report for this particular issue. I'll keep this thread open to keep you up to date on when this is resolved. Please let us know if you have any other questions or concerns in the meantime.

**Comment by bhkhatri221** [comment]
Hello @richardkeit,

A fix for this issue has been included in our [v1.6.0 release](https://github.com/awslabs/landing-zone-accelerator-on-aws/releases/tag/v1.6.0). Please update your global-config.yaml file and add s3ResourcePolicyAttachments. Example is provided in the [lza-sample-config](https://github.com/awslabs/landing-zone-accelerator-on-aws/tree/main/reference/sample-configurations/lza-sample-config).

 I will be closing this issue as resolved. Thanks for your support of Landing Zone Accelerator!

**Comment by yermulnik** [comment]
I'm wondering whether the referred S3 bucket policy has been validated? 🤔 

https://github.com/awslabs/landing-zone-accelerator-on-aws/blob/main/reference/sample-configurations/lza-sample-config/bucket-policies/central-log-bucket.json#L5
```json
      "Sid": "Allow Organization principals to put session manager logs",
```

Though https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_elements_sid.html
> The `Sid` element supports ASCII uppercase letters (A-Z), lowercase letters (a-z), and numbers (0-9).

TL;DR: spaces in `Sid` look worrisome
